src/m365/spo/commands/web/web-applicationcustomizer-add.spec.ts

```diff
@@ -0,0 +1,126 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import { telemetry } from '../../../../telemetry';
+import auth from '../../../../Auth';
+import { Cli } from '../../../../cli/Cli';
+import { CommandInfo } from '../../../../cli/CommandInfo';
+import { Logger } from '../../../../cli/Logger';
+import Command, { CommandError } from '../../../../Command';
+import { pid } from '../../../../utils/pid';
+import { sinonUtil } from '../../../../utils/sinonUtil';
+import commands from '../../commands';
+const command: Command = require('./web-applicationcustomizer-add');
+import * as SpoCustomActionAddCommand from '../customaction/customaction-add';
+
+
+describe(commands.WEB_APPLICATIONCUSTOMIZER_ADD, () => {
+  const webUrl = 'https://contoso.sharepoint.com';
+  const title = 'PageFooter';
+  const clientSideComponentId = '76d5f8c8-6228-4df8-a2da-b94cbc8115bc';
+  const clientSideComponentProperties = '';
+
+
+  let log: any[];
+  let logger: Logger;
+  let commandInfo: CommandInfo;
+  let loggerLogToStderrSpy: sinon.SinonSpy;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogToStderrSpy = sinon.spy(logger, 'logToStderr');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      Cli.executeCommand
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.WEB_APPLICATIONCUSTOMIZER_ADD), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('adds application customizer to a specific site without specifying clientSideComponentId', async () => {
+    sinon.stub(Cli, 'executeCommand').callsFake(async (command) => {
+      if (command === SpoCustomActionAddCommand) {
+        return;
+      }
+
+      throw 'Invalid request';
```

Comment:
This exception is odd since we're not executing a request but rather a command. Let's adjust it.

Reply:
Adjusted the error message. In the 'throws an error' test, I will throw an actual error that I've retrieved when executing the command.